cfssl_scan.go

```diff
@@ -0,0 +1,103 @@
+package main
+
+import (
+	"fmt"
+	"regexp"
+
+	"github.com/cloudflare/cfssl/scan"
+)
+
+var scanUsageText = `cfssl scan -- scan a host for issues
+
+Usage of scan:
+        cfssl scan [-short] [-scanfamily regexp] [-scanner regexp] HOST+
+        cfssl scan -list [-short] [-scanfamily regexp] [-scanner regexp]
+
+Arguments:
+        HOST:    Host(s) to scan (including port)
+Flags:
+`
+var scanFlags = []string{"short", "list", "family", "scanner"}
+var scanFamilyRegexp, scannerRegexp *regexp.Regexp
+
+func regexpLoop(familyFunc func(*scan.Family) error, scannerFunc func(*scan.Scanner) error) (err error) {
+	if Config.family != "" {
+		scanFamilyRegexp = regexp.MustCompile(Config.family)
+	}
+	if Config.scanner != "" {
+		scannerRegexp = regexp.MustCompile(Config.scanner)
+	}
+	for _, scanFamily := range scan.AllFamilies {
+		isMatch := true
+		if Config.family != "" {
+			isMatch = scanFamilyRegexp.MatchString(scanFamily.Name)
+		}
+		if isMatch {
+			if Config.scanner == "" {
+				err = familyFunc(scanFamily)
+				if err != nil {
+					return
+				}
+			}
+			for _, scanner := range scanFamily.Scanners {
+				if Config.scanner == "" || scannerRegexp.MatchString(scanner.Name) {
+					err = scannerFunc(scanner)
+					if err != nil {
+						return
+					}
+				}
+			}
+		}
+	}
+	return
+}
+func scanMain(args []string) (err error) {
```

Comment:
Most of the other cfssl Mains use `(err error)`, so I'll stick with that, but fix the `return err`/`return nil` instances